test/lit/passes/type-merging.wast

```diff
@@ -189,24 +324,31 @@
 ;; into $D. While doing so we must update the fields and the expressions that
 ;; they appear in, and not error.
 (module
-  ;; CHECK:      (type $C (struct (field (mut i32))))
-
-  ;; CHECK:      (type $D (struct_subtype (field (mut i32)) (field (mut i32)) $C))
-
-  ;; CHECK:      (type $I (array (mut (ref null $C))))
-  (type $I (array (mut (ref null $C))))
-  (type $C (struct (field (mut i32))))
-  (type $D (struct_subtype (field (mut i32)) (field (mut i32)) $C))
-  (type $E (struct_subtype (field (mut i32)) (field (mut i32)) $D))
-  (type $F (struct_subtype (field (mut i32)) (field (mut i32)) $E))
-  (type $D$to-merge (struct_subtype (field (mut i32)) (field (mut i32)) $F))
-  ;; CHECK:      (type $G (func (param (ref $C)) (result (ref $D))))
-  (type $G (func (param (ref $C)) (result (ref $D))))
-  ;; CHECK:      (type $H (struct_subtype (field (mut i32)) (field (mut i32)) (field (mut (ref null $D))) $D))
-  (type $H (struct_subtype (field (mut i32)) (field (mut i32)) (field (mut (ref null $E))) $D))
-  ;; CHECK:      (type $A (struct_subtype (field (mut i32)) (field (mut i32)) (field (mut (ref null $D))) (field (mut i64)) (field (mut (ref null $I))) $H))
-  (type $A (struct_subtype (field (mut i32)) (field (mut i32)) (field (mut (ref null $E))) (field (mut i64)) (field (mut (ref null $I))) $H))
-  (type $A$to-merge (struct_subtype (field (mut i32)) (field (mut i32)) (field (mut (ref null $E))) (field (mut i64)) (field (mut (ref null $I))) $A))
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $C (struct (field (mut i32))))
+
+    ;; CHECK:       (type $D (struct_subtype (field (mut i32)) (field (mut i32)) $C))
+
+    ;; CHECK:       (type $E (struct_subtype (field (mut i32)) (field (mut i32)) $D))
```

Comment:
I guess `$E` appears now when it didn't in the previous output because of the change to hybrid typing?

Reply:
Possibly, or it could be the bug that we're not properly replacing merged supertypes.